common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -126,4 +150,38 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
       }
     }
   }
+
+  private class DownloadCallback implements StreamCallback {
+
+    private WritableByteChannel channel = null;
+    private File targetFile = null;
+    private int chunkIndex;
+
+    public DownloadCallback(File targetFile, int chunkIndex) throws IOException {
+      this.targetFile = targetFile;
+      this.channel = Channels.newChannel(new FileOutputStream(targetFile));
```

Comment:
Pardon, I could hardly believe there are two `DownloadCallback`s download the same content to the same target file. In my understanding:
1. When `RetryingBlockFetcher` retry, there is no `DownloadCallback` initialized;
2. When failure of fetching chunk, retry from `RetryingBlockFetcher` will not be triggered.